src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,381 @@
+        GetWindowReorderingAncestors(Node startingNode, int maxSearchDepth, Array<NodePath>? reorderingNodePaths)
+    {
+        if (reorderingNodePaths == null || reorderingNodePaths.Count == 0)
+        {
+            return GetWindowReorderingAncestors(startingNode, maxSearchDepth);
+        }
+
+        return GetWindowReorderingAncestors(startingNode, reorderingNodePaths);
+    }
+
+    public override void _ExitTree()
+    {
+        connectionsEstablished = false;
+        windowReorderingAncestors.Clear();
+        connectedWindows.Clear();
+        connectedSiblings.Clear();
+        topSibling = null;
+    }
+
+    public void ConnectWindow(CustomDialog window, Node topNode, bool recursive)
+    {
+        // Make sure to be connected to other window reordering nodes before trying to connect the window
+        Setup();
+
+        if (recursive)
+        {
+            // Connection is recursive, connect to other window reordering nodes
+            foreach (var ancestor in windowReorderingAncestors)
+            {
+                ancestor.ReorderingNode.ConnectWindow(window, ancestor.Sibling, recursive);
+            }
+        }
+
+        if (window.IsConnected(nameof(CustomDialog.Dragged), this, nameof(OnWindowReorder)))
+        {
+            // This window is already connected here
+            GD.PrintErr($"A window {window.Name} ({window}) tried to connect to {Name} ({this}) multiple times");
+            return;
+        }
+
+        window.Connect(nameof(CustomDialog.Dragged), this, nameof(OnWindowReorder));
+
+        var binds = new Array();
+        binds.Add(window);
+        window.Connect(nameof(CustomWindow.Opened), this, nameof(OnWindowReorder), binds);
+
+        connectedWindows.Add(window, topNode);
+        connectedSiblings.Add(topNode);
+
+        // Update top sibling
+        if (topSibling == null || topSibling.GetIndex() < topNode.GetIndex())
+            topSibling = topNode;
+    }
+
+    public void DisconnectWindow(CustomDialog window, bool recursive)
+    {
+        if (recursive)
+        {
+            // Connection was recursive, disconnect from other window reordering nodes
+            foreach (var ancestor in windowReorderingAncestors)
+            {
+                ancestor.ReorderingNode.DisconnectWindow(window, recursive);
+            }
+        }
+
+        if (!window.IsConnected(nameof(CustomDialog.Dragged), this, nameof(OnWindowReorder)))
+        {
+            GD.PrintErr(
+                $"A window {window.Name}({window}) tried to disconnect from {Name}{this} but it wasn't connected");
+            return;
+        }
+
+        window.Disconnect(nameof(CustomDialog.Dragged), this, nameof(OnWindowReorder));
+        window.Disconnect(nameof(CustomWindow.Opened), this, nameof(OnWindowReorder));
+
+        var windowSibling = connectedWindows[window];
+        connectedWindows.Remove(window);
+
+        if (!connectedSiblings.Contains(windowSibling))
```

Comment:
I think this bit got so refactored that it doesn't make any sense anymore...

As far as I can see this would do absolutely nothing as this checks that something is not in a hash set and then it would remove it from the hash set, but the only way the check passes if it wasn't already in the hash set.

Reply:
I'll change this line to `if (connectedWindows.All(w => w.Value != windowSibling))` as I assume that's the intended check.

Reply:
Yes, that's the intended check. Also I think that I made this mistake in the beginning.